### PR TITLE
Expose `message-bytes-limit` in config

### PR DESCRIPTION
**Pull request by hendrikmakait**
Addresses
* https://github.com/dask/distributed/issues/7052#issuecomment-1258167521

Conflicts with
* #7071 (requires adjustments)

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: distributed/distributed-schema.yaml [hunk 1]
```diff
@@ -339,6 +339,17 @@ properties:
               To avoid this, Dask usually used a file-based lock.
               However, on some systems file-based locks don't work.
               This is particularly common on HPC NFS systems, where users may want to set this to false.
+          transfer:
```
Comment (hendrikmakait) [inline distributed/distributed-schema.yaml:342]: XREF: #6977 

I went with singular `transfer` to be coherent with variable naming.

File: distributed/worker_state_machine.py [hunk 2]
```diff
@@ -1129,7 +1130,7 @@ class WorkerState:
     #: :meth:`BaseWorker.gather_dep`. Multiple small tasks that can be fetched from the
     #: same worker will be clustered in a single instruction as long as their combined
     #: size doesn't exceed this value.
-    transfer_message_target_bytes: int
+    transfer_message_bytes_limit: float
```
Comment (hendrikmakait) [inline distributed/worker_state_machine.py:1133]: Driveby: Renamed to match other attribute names (e.g., `transfer_incoming_bytes_limit`). cc @crusaderky

**Comment by hendrikmakait** [comment]
CI flakes
* #6999

**Comment by crusaderky** [comment]
thank you!

File: distributed/distributed-schema.yaml [hunk 3]
```diff
@@ -339,6 +339,17 @@ properties:
...
+            type: object
+            description: |
+              Configuration setting for data transfer between workers
+            properties:
+              message-bytes-limit:
+                type:
+                  - string
+                  - integer
+                description: |
+                  The maximum size of a message sent between workers
```
[the remark this answers is not in the document]
Comment (hendrikmakait) [reply distributed/distributed-schema.yaml:352]: It does not explain the fine print, that's a fair point. I'll file a follow-up PR that highlights that this is not an absolute maximum.
